tools/traffic_vault_migrate/postgres.go

```diff
@@ -0,0 +1,743 @@
+package main
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"database/sql"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	util "github.com/apache/trafficcontrol/lib/go-util"
+
+	_ "github.com/lib/pq"
+)
+
+// PGConfig represents the configuration options available to the PG backend
+type PGConfig struct {
+	Host     string `json:"host"`
+	Port     string `json:"port"`
+	User     string `json:"user"`
+	Password string `json:"password"`
+	SSLMode  string `json:"sslmode"`
+	Database string `json:"database"`
+	Key      string `json:"aesKey"`
+	AESKey   []byte
+}
+
+// PGBackend is the Postgres implementation of TVBackend
+type PGBackend struct {
+	sslKey pgSSLKeyTable
+	dnssec pgDNSSecTable
+	uri    pgURISignKeyTable
+	url    pgURLSigKeyTable
+	cfg    PGConfig
+	db     *sql.DB
+}
+
+// String returns a high level overview of the backend and its keys
+func (pg *PGBackend) String() string {
+	data := fmt.Sprintf("PG server %v@%v:%v\n", pg.cfg.User, pg.cfg.Host, pg.cfg.Port)
+	data += fmt.Sprintf("\tSSL Keys: %v\n", len(pg.sslKey.Records))
+	data += fmt.Sprintf("\tDNSSec Keys: %v\n", len(pg.dnssec.Records))
+	data += fmt.Sprintf("\tURI Keys: %v\n", len(pg.uri.Records))
+	data += fmt.Sprintf("\tURL Keys: %v\n", len(pg.url.Records))
+	return data
+}
+
+// Name returns the name for this backend
+func (pg *PGBackend) Name() string {
+	return "PG"
+}
+
+// ReadConfig takes in a filename and will read it into the backends config
```

Comment:
It does take in a file name (both backends call `UnmarshalConfig` which calls `ioutil.ReadFile`. I've renamed this fn to `ReadConfigFile` and I'll also change `toCfg/fromCfg` to `toCfgPath/fromCfgPath` so that it's more clear what they contain.

Reply:
My bad -- I was reading `UnmarshalConfig` as `UnmarshalJSON`, so I thought `s` was actually the file's contents. Please disregard 😄